crates/native/src/media/track.rs

```diff
@@ -0,0 +1,565 @@
+            events_tx: None,
+            origin: track_origin,
+        };
+
+        res.add_video_sink(&mut sink);
+        res.kind.sink = Some(sink);
+
+        Ok(res)
+    }
+
+    /// Wraps the track of the `transceiver.receiver.track()` into a
+    /// [`VideoTrack`].
+    pub(crate) fn wrap_remote(
+        transceiver: &sys::RtpTransceiverInterface,
+        peer: &Arc<PeerConnection>,
+    ) -> Self {
+        let receiver = transceiver.receiver();
+        let track = receiver.track();
+        let track_origin = TrackOrigin::Remote(peer.id());
+
+        let dimensions = Arc::new(OnceLock::new());
+        dimensions
+            .set(RwLock::from(VideoDimensions { width: 0, height: 0 }))
+            .unwrap();
+        let mut sink = VideoSink::new(
+            i64::from(next_id()),
+            sys::VideoSinkInterface::create_forwarding(Box::new(
+                VideoFormatSink { dimensions: Arc::clone(&dimensions) },
+            )),
+            VideoTrackId(track.id()),
+            track_origin,
+        );
+
+        let mut res = Self {
+            kind: kind::Video {
+                id: VideoTrackId(track.id()),
+                inner: track.try_into().unwrap(),
+                sinks: Vec::new(),
+                dimensions,
+                sink: None,
+                // Safe to unwrap since transceiver is guaranteed
+                // to be negotiated at this point.
+                source: MediaTrackSource::Remote {
+                    mid: transceiver.mid().unwrap(),
+                    peer: Arc::downgrade(peer),
+                },
+            },
+            senders: HashMap::new(),
+            events_tx: None,
+            origin: track_origin,
+        };
+
+        res.add_video_sink(&mut sink);
+        res.kind.sink = Some(sink);
+
+        res
+    }
+
+    /// Adds the provided [`VideoSink`] to this [`VideoTrack`].
+    pub fn add_video_sink(&mut self, video_sink: &mut VideoSink) {
+        self.kind.add_video_sink(video_sink);
+    }
+
+    /// Detaches the provided [`VideoSink`] from this [`VideoTrack`].
+    pub fn remove_video_sink(&mut self, video_sink: VideoSink) {
+        self.kind.remove_video_sink(video_sink);
+    }
+}
+
+impl From<&VideoTrack> for api::MediaStreamTrack {
+    fn from(track: &VideoTrack) -> Self {
+        Self {
+            id: track.id().0,
+            device_id: match track.source() {
+                MediaTrackSource::Local(src) => src.device_id().to_string(),
+                MediaTrackSource::Remote { .. } => "remote".into(),
+            },
+            kind: api::MediaType::Video,
+            enabled: true,
+            peer_id: match track.origin {
+                TrackOrigin::Local => None,
+                TrackOrigin::Remote(peer_id) => Some(peer_id.into()),
+            },
+        }
+    }
+}
+
+/// Representation of an `Audio` track interface.
+pub type AudioTrack = Track<kind::Audio>;
+
+impl AsRef<sys::AudioTrackInterface> for AudioTrack {
```

Comment:
Most of things like that could be implemented with a `derive_more` crate. We do not need `AsRef` now, but try to impl those things with macro in the future.